include/musica/micm.hpp

```diff
@@ -39,8 +39,10 @@ namespace musica
     /// @brief Types of MICM solver
     enum MICMSolver
     {
-      Rosenbrock = 1,           // Vector-ordered Rosenbrock solver
-      RosenbrockStandardOrder,  // Standard-ordered Rosenbrock solver
+      Rosenbrock = 1,              // Vector-ordered Rosenbrock solver
+      BackwardEuler,               // Vector-ordered BackwardEuler solver
+      RosenbrockStandardOrder,     // Standard-ordered Rosenbrock solver
+      BackwardEulerStandardOrder,  // Standard-ordered BackwardEuler solver
     };
```

Comment:
The change of the enum order will fail the fortran micm APIs. I guess the ordering based on the matrix type is fine as it is because we only have two types of solver, but if we think about adding more solvers, it'd probably be easier for maintainers to order them based on the solver type, then the enum value doesn't have to change every time we add a new solver. Also for the users, if a new release has to change the ordering, it could be confusing.

Reply:
I agree it will better to order by solver type / matrix type ... will also rearrange source with this ordering.